lib/Varien/Object.php

```diff
@@ -318,56 +320,61 @@ public function getData($key = '', $index = null)
-                }
-                if (is_array($data)) {
-                    if (!isset($data[$k])) {
-                        return $default;
-                    }
-                    $data = $data[$k];
-                } elseif ($data instanceof Varien_Object) {
-                    $data = $data->getData($k);
-                } else {
-                    return $default;
-                }
-            }
-            return $data;
+        $data = $this->_data[$key] ?? null;
+        if ($data === null && $key !== null && strpos($key, '/') !== false) {
+            /* process a/b/c key as ['a']['b']['c'] */
+            $data = $this->getDataByPath($key);
         }
 
-        // legacy functionality for $index
-        if (isset($this->_data[$key])) {
-            if (is_null($index)) {
-                return $this->_data[$key];
+        if ($index !== null) {
+            if ($data === (array)$data) {
+                $data = $data[$index] ?? null;
+            } elseif (is_string($data)) {
+                $data = explode(PHP_EOL, $data);
+                $data = $data[$index] ?? null;
+            } elseif ($data instanceof Varien_Object) {
+                $data = $data->getData($index);
+            } else {
+                $data = null;
             }
+        }
+        return $data;
+    }
+
+    /**
+     * Get object data by path
+     *
+     * Method consider the path as chain of keys: a/b/c => ['a']['b']['c']
+     *
+     * @param string $path
+     * @return mixed
+     */
+    public function getDataByPath($path)
+    {
+        $keys = explode('/', (string)$path);
 
-            $value = $this->_data[$key];
-            if (is_array($value)) {
-                //if (isset($value[$index]) && (!empty($value[$index]) || strlen($value[$index]) > 0)) {
-                /**
-                 * If we have any data, even if it empty - we should use it, anyway
-                 */
-                if (isset($value[$index])) {
-                    return $value[$index];
-                }
+        $data = $this->_data;
+        foreach ($keys as $key) {
+            if ((array)$data === $data && isset($data[$key])) {
+                $data = $data[$key];
+            } elseif ($data instanceof Varien_Object) {
+                $data = $data->getDataByKey($key);
+            } else {
                 return null;
-            } elseif (is_string($value)) {
-                $arr = explode("\n", $value);
-                return (isset($arr[$index]) && (!empty($arr[$index]) || strlen($arr[$index]) > 0))
-                    ? $arr[$index] : null;
-            } elseif ($value instanceof Varien_Object) {
-                return $value->getData($index);
             }
-            return $default;
         }
-        return $default;
+        return $data;
+    }
+
+    /**
+     * Get object data by particular key
+     *
+     * @param string $key
+     * @return mixed
+     */
+    public function getDataByKey($key)
```

Comment:
Only accept `string` ?

Reply:
Yes, `$key` is expected to be of type string.